### PR TITLE
migrate api/availability/[user] to viewer.availability.user in trpc

### PR DESCRIPTION
**Pull request by hussamkhatib**
## What does this PR do?

- change API route from `/api/availability/[user]`
to trpc query -> `viewer.availability.user`
-Refs: #3573

- There is a non-null assertion `user.username!` (is that okay? I tried to fix it, but at the of the username is typed like that `string | null` )
- This [api route](https://github.com/calcom/cal.com/blob/main/apps/web/pages/api/availability/%5Buser%5D.ts) is not deleted yet. Reason : It is still being used by[useSlots](https://github.com/calcom/cal.com/blob/main/apps/web/lib/hooks/useSlots.ts) hook and this useSlot hook is not used anywhere.
- function `convertMinsToHrsMins`  is moved out of component since it is a pure function and we dont want to create new instance of it for every re-render
**Environment**: Staging(main branch) / Production

## How should this be tested?

>  Please describe the tests that you ran to verify your changes. 
Manual testing did the job.
> Provide instructions so we can reproduce.Please also list any relevant details for your test configuration
- apply the changes to your local developement, navigate to `http://localhost:3000/availability/troubleshoot` , compare data returned by new and old method. 

PS: This is my time contributing to code here, is this information provided in PR description enough ?
What all kind of tests are ran creating a PR, I could see e2e and lint in contribution guildelines

### CODE REVIEW

File: apps/web/pages/availability/troubleshoot.tsx [hunk 1]
```diff
@@ -13,40 +13,16 @@ type User = inferQueryOutput<"viewer.me">;
...
-        .finally(() => {
-          setLoading(false);
-        });
-    };
-    fetchAvailability(selectedDate);
-  }, [user.username, selectedDate]);
+  const { data, isLoading } = trpc.useQuery([
+    "viewer.availability.user",
+    {
+      user: user.username!,
```
Comment (zomars) [inline apps/web/pages/availability/troubleshoot.tsx:21]: Let's take advantage of the change to correct this input naming:
```suggestion
      username: user.username!,
```

File: apps/web/pages/availability/troubleshoot.tsx [hunk 2]
```diff
@@ -13,40 +13,16 @@ type User = inferQueryOutput<"viewer.me">;
...
+      dateFrom: selectedDate.startOf("day").utc().format(),
+      dateTo: selectedDate.endOf("day").utc().format(),
+    },
+  ]);
```
Comment (zomars) [inline apps/web/pages/availability/troubleshoot.tsx:25]: Also we can be sure that we only fetch when the username is available

```suggestion
  const { data, isLoading } = trpc.useQuery(
    [
      "viewer.availability.user",
      {
        username: user.username!,
        dateFrom: selectedDate.startOf("day").utc().format(),
        dateTo: selectedDate.endOf("day").utc().format(),
      },
    ],
    { enabled: !!user.username }
  );
```

File: packages/trpc/server/routers/viewer/availability.tsx [hunk 3]
```diff
@@ -93,6 +95,23 @@ export const availabilityRouter = createProtectedRouter()
       };
     },
   })
+  .query("user", {
+    input: z.object({
+      user: z.string(),
```
Comment (zomars) [inline packages/trpc/server/routers/viewer/availability.tsx:100]:
```suggestion
      username: z.string(),
```

To match the comment above

**Review by zomars — COMMENTED** [review]
Thank you for your patience and very nice contribution. Let's address some quick comments before merging 🙏🏽

**Comment by hussamkhatib** [comment]
Thanks, I have resolved all the changes requested.

**Review by emrysal — APPROVED** [review]

**Pull request merged**